packages/dm-core/src/components/NewEntityButton.tsx

```diff
@@ -143,21 +146,25 @@ export function NewEntityButton(props: {
             )}
             <Button
               disabled={
-                !(newName && saveDestination && (typeToCreate || copyTarget))
+                !(
+                  newName &&
+                  saveDestination &&
+                  (typeToCreate || documentToCopy)
+                )
               }
               type="submit"
               onClick={() => {
                 setLoading(true)
 
-                if (copyTarget) {
+                if (documentToCopy) {
                   // @ts-ignore
-                  delete copyTarget._id
-                  copyTarget.name = newName
+                  delete documentToCopy._id
+                  documentToCopy.name = newName
```

Comment:
Shouldn't state variables be immutable? I don't think it will matter in this case, but as a principle it would be better to make a copy
```suggestion
                  const newDocument = { ...documentToCopy, name: newName }
                  delete newDocument._id
```
Also, I don't think the @ts-ignore is doing anything anymore

Reply:
agree! Will fix